geode-core/src/main/java/org/apache/geode/management/internal/operation/OperationHistoryManager.java

```diff
@@ -96,10 +96,15 @@ private static boolean isExpired(long expirationTime, OperationState<?, ?> opera
   }
 
   private OperationState<?, ?> validateLocator(OperationState<?, ?> operationState) {
+    if (operationState.getOperationEnd() != null) {
+      return operationState;
+    }
     if (isLocatorOffline(operationState)) {
       operationState.setOperationEnd(new Date(), null,
           new RuntimeException("Locator that initiated the Rest API operation is offline: "
               + operationState.getLocator()));
+      operationStateStore.recordEnd(operationState.getId(), operationState.getResult(),
```

Comment:
I think you should call "recordEnd" here instead of "operationStateStore.recordEnd". Also it looks like recordEnd will in turn call operationState.setOperationEnd. So I think this whole if block could just be this one call: recordEnd(new Date(), null,
          new RuntimeException("Locator that initiated the Rest API operation is offline: "
              + operationState.getLocator()));

Reply:
Changes are made as suggested. And the method name is changed to reflect the value is recorded.